independent-projects/tools/devtools-common/src/main/java/io/quarkus/registry/builder/RegistryBuilder.java

```diff
@@ -119,4 +119,19 @@ public Registry build() {
         String getQuarkusVersion();
     }
 
+    @Override
+    public void visitExtensionRelease(String groupId, String artifactId, Release release) {
+        ArtifactKey extensionKey = ImmutableArtifactKey.of(groupId, artifactId);
+        ArtifactCoords coords = ImmutableArtifactCoords.of(extensionKey, release.getVersion());
+        ArtifactCoordsTuple key = ImmutableArtifactCoordsTuple.builder().coords(coords)
+                .quarkusVersion(release.getQuarkusCore())
+                .build();
+        ModifiableExtensionRelease releaseBuilder = releases.computeIfAbsent(key,
+                appArtifactCoords -> ModifiableExtensionRelease.create()
+                        .setRelease(Release.builder()
+                                .version(appArtifactCoords.getCoords().getVersion())
+                                .quarkusCore(appArtifactCoords.getQuarkusVersion())
+                                .build()));
+        releaseBuilder.setRelease(release);
```

Comment:
This method is quite similar to the `visitExtension(Extension,String)` one. Wondering if we can change the original `visitExtension` method to accept a `Release` instead of `String` to reuse the method?

Reply:
Also, I fail to see where `getCompatibleQuarkusCore` is called in this PR

Reply:
It's called in my branch.